gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -650,6 +809,40 @@ gst_tensor_filter_common_set_property (GstTensorFilterPrivate * priv,
         priv->is_updatable = g_value_get_boolean (value);
       break;
     }
+    case PROP_INPUTLAYOUT:
+      g_assert (!prop->input_configured && value);
+      /* Once configured, it cannot be changed in runtime */
```

Comment:
No. it may be changed (from NULL to valid value). With dynamic pipelines, you don't know when the framework (Gst) will set the value. It may repeatedly call with the same  property value or value->NULL->value.

Reply:
Updated. Now the data layout can be updated as long as the framework is not yet opened.
Changing the data layout after the framework is already opened will be supported later.